cypress/support/e2e.js

```diff
@@ -31,20 +32,24 @@ before(() => {
       localeSettingsId = body.id;
       localeSettingsValue = body.value;
       const localeValue = localeSettingsValue.match(/"locale":"(.*?)"/);
-      if (localeValue) {
-        const extractedLocale = localeValue[1];
-        Cypress.env('localeValue', extractedLocale);
-        if (extractedLocale !== defaultTestLanguage) {
-          const updatedValue = localeSettingsValue.replace(`"locale":"${extractedLocale}"`, `"locale":"${defaultTestLanguage}"`);
-          cy.putLocaleSettings(localeSettingsId, updatedValue);
-        }
+      const timezoneValue = localeSettingsValue.match(/"timezone":"(.*?)"/);
+      const extractedLocale = localeValue ? localeValue[1] : undefined;
+      const extractedTimezone = timezoneValue ? timezoneValue[1] : undefined;
+
+      Cypress.env('localeValue', extractedLocale);
+      Cypress.env('timezoneValue', extractedTimezone);
+
+      if (extractedLocale !== defaultTestLanguage || extractedTimezone !== defaultTestTimezone) {
+        const updatedValue = localeSettingsValue.replace(/"locale":"[^"]*"/, `"locale":"${defaultTestLanguage}"`).replace(/"timezone":"[^"]*"/, `"timezone":"${defaultTestTimezone}"`);
+        cy.putLocaleSettings(localeSettingsId, updatedValue);
       }
     }
   });
 });
 
 after(() => {
-  if (Cypress.env('localeValue') && Cypress.env('localeValue') !== defaultTestLanguage) {
+  if ((Cypress.env('localeValue') && Cypress.env('localeValue') !== defaultTestLanguage) ||
```

Comment:
Same comment as above, does this always work with an OR or do we need two statements?

Reply:
Yes.
In fact I had 2 statements first and this was tricky because `localeSettingsValue` was not always updated twice if both values were to be replaced.
Probably also something with async testing...